src/types.ts

```diff
@@ -145,7 +145,7 @@ export type MutatorCallback<Data = any> = (
 export type MutatorOptions<Data = any> = {
   revalidate?: boolean
   populateCache?: boolean | ((result: any, currentData: Data) => Data)
-  optimisticData?: Data
+  optimisticData?: Data | ((data?: Data) => Data)
```

Comment:
```suggestion
  optimisticData?: Data | ((currentData?: Data) => Data)
```

Let's name this as `currentData` so TypeScript will give better intellisense.